docs/application/dotnet/guides/nui/styling-controls-with-xaml.md

```diff
@@ -0,0 +1,182 @@
+
+The NUI also defines the ViewStyle derived classes for components:  `ImageView`, `TextLabel` and `TextField`
+
+```csharp
+ImageViewStyle swtichImageStyle = new ImageViewStyle
+{
+    ResourceURL = new StringSelector
+    {
+        Normal = "controller_switch_bg_off.png",
+        Selected = "controller_switch_bg_on.png",
+        Disabled = "controller_switch_bg_off_dim.png",
+        DisabledSelected = "controller_switch_bg_on_dim.png"
+    }
+};
+
+TettLabelStyle redTextStyle = new TextLabelStyle()
+{
+  TextColor = Color.Red,
+  PointSize = 20
+};
+
+TextFieldStyle textStyle = new TextFieldStyle()
+{
+  EnableCursorBlink = False
+};
+```
+
+# XAML themes
+
+In more advanced applications it is better to use XAML Theme files to define UI style for NUI widgets. With XAML files it is easier to share styles between different applications. From the point of view of the application architecture, XAML allows for better separation of view definitions from application logic.
+
+## XAML Example
+
+Basic text style defined in previous section could be implemented using XAML in the following way:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<Theme
+  xmlns="http://tizen.org/Tizen.NUI/2018/XAML"
+  xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml">
+
+  <TextLabelStyle x:Key="redTextStyle" TextColor="Red" PointSize="20" />
+</Theme>
+```
+
+## XAML states
+
+Some properties in ViewStyle are Selector types that can have multiple values by control's state.
+
+```csharp
+Selector selector = new Selector<Color>()
+{
+  Normal = Color.Black,
+  Pressed = Color.Blue,
+  Disabled = Color.Red
+};
+```
+
+The same xaml code is,
+
+```xml
+<Selector x:TypeArguments="Color" Normal="Black" Pressed="Blue" Disabled="Red"/>
+```
+
+Selectors works when the control state is enabled in a View. Control state are enabled for all controls in Tizen.NUI.Components by default, but not for those in Tizen.NUI.BaseComponents, such as View and TextLabel. If you want to enable control state for them, please set EnableControlState property to true.
+
+```xml
+<ViewStyle x:Key="viewColoredByTouch" EnableControlState="true">
+  <ViewStyle.BackgroundColor>
+      <Selector x:TypeArguments="Color" Normal="Black" Pressed="Blue" Disabled="Red"/>
+  </ViewStyle.BackgroundColor>
+</ViewStyle>
+```
+}
+
+## Load theme in NUI application
+
+```csharp
+using Tizen.NUI; 
+
+namespace NUI_Theme
+{
+  class Program : NUIApplication
+  {
+    protected override void OnCreate()
+    {
+      // Load theme from the xaml file
+      Theme theme = new Theme("full_path_of_the_xaml_file.xaml");
+
+      // Apply it to the current NUI Application
+      ThemeManager.ApplyTheme(theme);
+
+      // Set style name in NUI widget
+      TextLabel textLabel = new TextLabel()
+      textLabel.StyleName = "redTextStyle";
+    }
+  }
+```
+
+## Handle theme changes in runtime
+
+Two methods can be used to handle theme changes:
+* `ThemeManager.ThemeChanged`
+* `View.OnThemeChanged`
```

Comment:
The following methods can be used to handle theme changes in runtime:
- `ThemeManager.ThemeChanged`
- `View.OnThemeChanged`